src/oop/static_modifier/hard/q2/MainTest.java

```diff
@@ -0,0 +1,188 @@
+        assertTrue(testClass.hasMethod(setAttributeName, new Class<?>[]{int.class}, Void.TYPE, new String[]{"public"}),
+                String.format(wrongModifierMessage, setAttributeName, className));
+        Object[][] arguments = {
+                {"value1", String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {value2, int.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        testClass.callMethod(setAttributeName, setSizeArguments, setMethodModifiers, classInstance);
+        _assertEquals(value2, testClass.getFieldValue(null, staticAttributeName), incorrectSetMethodMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependency1")
+    @Order(5)
+    public void lightbulbClassHasCorrectToStringMethod(String value1, int value2) throws Throwable {
+        String missingMethodMessage = "The %s class is missing the %s method. Make sure that the class contains the method and it is spelt correctly with the correct arguments as parameters.";
+        String wrongTypeMessage = "The %s method in the %s class has the wrong return type.";
+        String wrongModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        assertTrue(testClass.hasMethod("toString", null),
+                String.format(missingMethodMessage, className, "toString"));
+        assertTrue(testClass.hasMethod("toString", null, String.class),
+                String.format(wrongTypeMessage, "toString", className));
+        assertTrue(testClass.hasMethod("toString", null, String.class, new String[]{"public"}),
+                String.format(wrongModifierMessage, "toString", className));
+        Object[][] arguments = {
+                {value1, String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        testClass.setFieldValue(null, value2, staticAttributeName);
+        String[] methodModifiers = {"public"};
+        String expected = "This lightbulb is a " + value1 + " light with " + value2 + " voltage.";
+        String incorrectToStringMessage = String.join(" ",
+                "Your", className, "toString method does not return the correct String.");
+        Object output = testClass.callMethod("toString", methodModifiers, classInstance);
+        _assertEquals(expected, output, incorrectToStringMessage);
+    }
+
+    @Test
+    @Tag("dependent1")
+    @Order(6)
+    public void correctTestClass() throws Throwable {
+        Clause[] clauses = {
+                new StringLiteral("This lightbulb is a LED light with 150 voltage."),
+                new NewLine(),
+                new StringLiteral("This lightbulb is a LCD light with 200 voltage."),
+                new NewLine(),
+                new StringLiteral("This lightbulb is a Halogen light with 300 voltage."),
+                new NewLine(),
```

Comment:
test values should be kept at the top of the test class as variables for ease of varying on the website